tests/meltano/core/state_store/test_db.py

```diff
@@ -31,47 +32,47 @@ def test_set_state(self, subject: DBStateStoreManager) -> None:
             state_id="partial_only",
             partial_state={"singer_state": {"partial": 1}},
         )
-        subject.set(partial_only)
+        subject.set(copy.deepcopy(partial_only))
         assert subject.get(partial_only.state_id) == partial_only
 
         # Partial merges onto existing partial
         new_partial = JobState(
             state_id="partial_only",
             partial_state={"singer_state": {"partial": 2}},
         )
-        subject.set(new_partial)
+        subject.set(copy.deepcopy(new_partial))
         assert subject.get(partial_only.state_id) == new_partial
```

Comment:
Because `merge(src, dst)` mutates `dst` in place, theses tests were tautological. Meaning that the `new_partial.partial_state` dictionary was mutated and made equal to the new partial state if the merge order was incorrect. Luckily, the merge order was correct but theses changes will ensure we catch any errors in the future.

Reply:
Using `copy.deepcopy` was useful to validate the changes to `DBStateStoreManager.set`, but I can revert this before merging.